src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingSubmissionService.java

```diff
@@ -147,6 +154,7 @@ public ProgrammingSubmission notifyPush(Long participationId, Object requestBody
             // as the VCS-server performs the request
             SecurityUtils.setAuthorizationObject();
 
+            // TODO: is this still allowed for an exam? what do we want to do here?
```

Comment:
theoretically in a very rare edge case this would be allowed for exam exercises

Reply:
ok we will handle this one in the follow-up PR